src/XMakeBuildEngine/Instance/ProjectInstance.cs

```diff
@@ -1817,6 +1819,15 @@ internal bool Build(string[] targets, IEnumerable<ILogger> loggers, IEnumerable<
             if (loggers != null)
             {
                 parameters.Loggers = (loggers is ICollection<ILogger>) ? ((ICollection<ILogger>)loggers) : new List<ILogger>(loggers);
+
+                foreach (var logger in loggers)
+                {
+                    if (logger.Verbosity == LoggerVerbosity.Diagnostic)
+                    {
+                        parameters.LogTaskInputs = true;
```

Comment:
I think this might be clearer as something like
```C#
parameters.LogTaskInputs = parameters.LogTaskInputs || loggers.Any(l => l.Verbosity == LoggerVerbosity.Diagnostic);
```

That is a bit clearer about the intent. It still probably deserves a comment, though.